pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -246,6 +246,20 @@ func (c *Reconciler) reconcile(ctx context.Context, tr *v1alpha1.TaskRun) error
 			Message: ReasonRunning,
 		})
 	}
+	tr.Status.StartTime = build.Status.StartTime
+	if build.Status.Cluster != nil {
+		tr.Status.PodName = build.Status.Cluster.PodName
+	}
+	tr.Status.CompletionTime = build.Status.CompletionTime
+
+	tr.Status.Steps = []v1alpha1.StepState{}
+	for i := 0; i < len(build.Status.StepStates); i++ {
+		state := build.Status.StepStates[i]
+		tr.Status.Steps = append(tr.Status.Steps, v1alpha1.StepState{
+			ContainerState: *state.DeepCopy(),
+			// FIXME(vdemeester) logsURL
+		})
+	}
```

Comment:
this feels to me like a chunk of code that could benefit from being in a separate function with unit tests - which would also let us significantly simplify the logic being added to taskrun_test - ill add a comment there too :)